sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -14,12 +14,8 @@ namespace Azure.Core
     /// </summary>
     public sealed class HttpMessage : IDisposable
     {
-        /// <summary>
-        /// This dictionary is keyed with <c>Type</c> for a couple of reasons. Primarily, it allows values to be stored such that even if the accessor methods
-        /// become public, storing values keyed by internal types make them inaccessible to other assemblies. This protects internal values from being overwritten
-        /// by external code. See the <see cref="TelemetryDetails"/> and <see cref="UserAgentValueKey"/> types for an example of this usage.
-        /// </summary>
-        private Dictionary<Type, object>? _typeProperties;
+        private Dictionary<string, object>? _properties;
```

Comment:
Related to @JoshLove-msft's comment - is the goal to remove this entirely?

Reply:
Not remove, just improve the performance of the implementation.